src/Form/Control/Calendar.php

```diff
@@ -145,26 +145,26 @@ public function getJsInstance(): JsExpression
         return (new Jquery('#' . $this->id . '_input'))->get(0)->_flatpickr;
     }
 
-    private function translateFormat(string $format): string
+    public static function translateFormat(string $format): string
```

Comment:
is there any reason for static except "it does not require `$this` and maybe is expected to not change"?

Reply:
SopeBuilder also uses flatpickr and I uses it to properly set options to it.

```
$options['dateComponent']['time_24hr'] = Calendar::use24hrTimeFormat($format);
$options['dateComponent']['noCalendar'] = ($type === 'time');
$options['dateComponent']['enableSeconds'] = Calendar::useSeconds($format);
```

Reply:
instantiate dummy Calendar control class in ScopeBuilder or UI Persistence if needed

Reply:
and why not use static function? Options are not going to change. if they do, they will need to be change everywhere...

Reply:
Static anything is not optimal from OOP design in general as DI is not easy then. I understand, that here you expect it will not be required to change, but what if user want to use different Calendar lib? I am very happy, that Atk does not use static, so I want to keep that clean code.

Reply:
make sense. I reverted it to public function instead.